clients/common/cblas_interface.cpp

```diff
@@ -109,6 +110,32 @@ extern "C" {
 
     //axpy
     template<>
+    void cblas_axpy<rocblas_half>( rocblas_int n,
+                            rocblas_half alpha,
+                            rocblas_half *x, rocblas_int incx,
+                            rocblas_half *y, rocblas_int incy)
+    {
+        rocblas_int abs_incx = incx >=0 ? incx : -incx;
+        rocblas_int abs_incy = incy >=0 ? incy : -incy;
+        float alpha_float = static_cast<float>(alpha);
+        std::unique_ptr<float []> x_float(new float[n*abs_incx]());
```

Comment:
`std::array<>` requires the size to be known at compile-time. For runtime sizes a `std::vector` can be used, such as:

```cpp
std::vector<float> x_float(n*abs_incx);
```

But this will store the size in `std::vector`, so it takes extra space than the `unique_ptr`. Most of the time, its not a problem.